gpiozero/boards.py

```diff
@@ -2102,11 +2102,16 @@ def value(self):
         """
         Returns :data:`True` if the socket is on and :data:`False` if the
         socket is off.  Setting this property changes the state of the socket.
+        Returns :data:`None` only when constructed with :data:`initial_value`
+        set to :data:`None` and neither :data:`on()` nor :data:`off()` have
+        been called since construction.
         """
         return self._value
 
     @value.setter
     def value(self, value):
+        if value is None:
+            raise TypeError('value cannot be None')
```

Comment:
A more specific exception would be better - hmm perhaps `EnergenieBadInitialValue` in `exc.py` could be renamed to `EnergenieBadValue` (with the detail-string updated too) and then used here? :slightly_smiling_face: